completions/ccache

```diff
@@ -26,8 +29,11 @@ _ccache()
 
     $split && return
 
-    COMPREPLY=( $( compgen -W '$( _parse_help "$1" )' -- "$cur" ) )
-    [[ $COMPREPLY == *= ]] && compopt -o nospace
+    if [[ "$cur" == -* ]]; then
+        COMPREPLY=( $( compgen -W '$( _parse_help "$1" )' -- "$cur" ) )
+        [[ $COMPREPLY == *= ]] && compopt -o nospace
+        return
+    fi
```

Comment:
I tried without this, and completion of the compiler's options seem to work just fine. I mean e.g. `ccache gcc -<TAB>` completes gcc's options, both with and without this change. The test suite also passes with or without it. So I still cannot find a justification for this part of the change. Could you give me a specific reproducer what exactly does not work with it?